### DISCUSSION TITLE
headers not set

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#3772] [Q&A] [answered]
```

useEffect(() => {
  const fetchData = async () => {
  
    const response = await pb.collection('Eligibility').getFullList();
    const result = await pb.collection('catview').getList(1, 500, {
      headers: {
        'time_stamp': timestamp,
        'created': timestamps.modifiedTimestamp,
        'api_key': api_key
      },
      expand: 'Factory,Warehose,brokersID,bidder_current,bidder_current.reference',
      filter: `brokersID="${pb.authStore.model.id}" && Season = "${response[0].Season}" &&  Sale_number="${parseInt(response[0].Sale_Number)}"  && Status = true  `,
      sort: '+created'
    });

  
  };

  if (api_key) {
    fetchData();
  }
}, [api_key, timestamp, timestamps.modifiedTimestamp]);


```
`http://127.0.0.1:8090/api/collections/posts/records?page=1&perPage=500&headers={"time_stamp":"1700233281795","created":"2023-11-17 15:01:23.795Z","api_key":"68391d4781cc211ac5b1133cfa1e3760f45f8ca39667dc3259c662624b0f567a"}  `

request are going like this how to fix that

installed pocketbase version "pocketbase": "^0.19.0",

**Comment by ganigeorgiev** [comment]
I suspect that you are not using JS SDK v0.19.0.

Double check that you use JS SDK v0.17.0+ and that there is no cache of your `node_modules`. If it still doesn't work, you can try to delete your `node_modules` and run `npm install` again.

If you use Next.js make sure that cache is disabled (_Next.js has custom global `fetch` implementation that comes with cache enabled by default_).


In any case I'm not able to reproduce it locally (_the JS SDK also has automated tests that should ensure the correct behavior_). Here is an example with a plain static `pb_public/index.html` file:

```html
<!-- pb_public/index.html -->
<!DOCTYPE html />
<html>
    <head>
        <meta charset="utf-8" />
        <title>Test custom options</title>
    </head>
    <body>
        <script type="module">
            import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@0.19.0/dist/pocketbase.es.mjs"

            const pb = new PocketBase("http://127.0.0.1:8090");

            pb.authStore.clear();

            await pb.collection("demo").getList(1, 20, {
                headers: {
                    'x-test1': "a",
                    'x-test2': "b",
                    'x-test3': "c"
                },
                expand: 'something',
                filter: `1=1`,
                sort: '+created'
            })
        </script>
    </body>
</html>
```

The above produces the following request showing that all parameters are properly constructed and send with the `fetch` call:

![request](https://i.imgur.com/5GmnJ3b.png)


_**Side-note: please don't open several related discussions and instead try to provide a follow up in your existing one https://github.com/pocketbase/pocketbase/discussions/3769 since it has a better chance to help others understanding the background of the question.**_

**Comment by breakingbad45** [comment]
i am using react js and iam trying to reinstall the package using yarn add pocketbase but still same issue

**Reply by ganigeorgiev** [reply, marked as answer]
I'm not sure how to help you. To rule out PocketBase issue please copy the above file in  `pb_public/index.html` and open `http://127.0.0.1:8090` in the browser. The request can be inspected via the browser's _Inspect Tool > Network_ tab.

If you think that this is a PocketBase issue, feel free to create a **minimal** reproducible repo OR at least provide some steps to reproduce, and I'll try to investigate it further.

**But again, please don't open multiple discussions for the same seemingly related thing and instead consider doing a followup with more details in this or one of your previous posts.**
